src/lib/mongoBackend/dateExpiration.cpp

```diff
@@ -0,0 +1,198 @@
+
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* processDateExpirationAtEntityCreation -
+*
+* This function process the context attribute vector, searching for an attribute meaning
+* date expiration for the entity. In that case, it fills dateExpiration value.
+* If a date expiration attribute is not found dateExpiration keeps empty value set by the caller function.
+*
+* This function always return true (no matter if the attribute was found or not), except in an
+* error situation, in which case errorDetail is filled.
+*/
+bool processDateExpirationAtEntityCreation
+(
+  const ContextAttributeVector&  caV,
+  mongo::Date_t*                 dateExpiration,
+  std::string*                   errDetail,
+  OrionError*                    oe
+)
+{
+  for (unsigned ix = 0; ix < caV.size(); ++ix)
+  {
+    const ContextAttribute* caP = caV[ix];
+
+    if (caP->name != DATE_EXPIRES)
+    {
+      continue;
+    }
+
+    if (!getDateExpiration(caP, dateExpiration, errDetail))
+    {
+      oe->fill(SccBadRequest, *errDetail, "BadRequest");
+      return false;
+    }
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* processDateExpirationAtUpdateAttribute -
+*/
+bool processDateExpirationAtUpdateAttribute
+(
+  const ContextAttribute*  targetAttr,
+  mongo::Date_t*           dateExpiration,
+  bool*                    replaceDate,
+  std::string*             errDetail,
+  OrionError*              oe
+)
+{
+  /*
+   * If the name of the target attribute is the date expiration,
+   * check for a number value that will be used in the mongo::Date_t constructor.
+   * If it is an empty value, is interpreted as a date expiration deletion and the ACTUAL date value
+   * is set to 0, in order to signal the caller function.
+   * If valid value, also replaceDate boolean is set to true, in order to manage the new date value
+   * in case the update is a replace operation.
+   */
+  if (targetAttr->name == DATE_EXPIRES)
+  {
+    if (targetAttr->numberValue)
+    {
+      if (!getDateExpiration(targetAttr, dateExpiration, errDetail))
+      {
+        oe->fill(SccBadRequest, *errDetail, "BadRequest");
+        return false;
+      }
+      else
+      {
+        *replaceDate = true;
+      }
+    }
+    else
+    {
+      *dateExpiration = 0;
+    }
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* processDateExpirationAtAppendAttribute -
+*/
+bool processDateExpirationAtAppendAttribute
+(
+  mongo::Date_t*                 dateExpiration,
+  const ContextAttribute*        targetAttr,
+  bool                           actualAppend,
+  std::string*                   errDetail,
+  OrionError*                    oe
+)
+{
+  if (targetAttr->numberValue)
+  {
+  if (!getDateExpiration(targetAttr, dateExpiration, errDetail))
```

Comment:
indentation of 2 spaces seems to be missing

Reply:
Fixed in abdaa28